staging/src/k8s.io/apiserver/pkg/server/filters/waitgroup.go

```diff
@@ -73,7 +75,17 @@ func withWaitGroup(handler http.Handler, longRunning apirequest.LongRunningReque
 			return
 		}
 
+		if terminator != nil && longRunningRequest {
+			newCtx, cancelFn := terminator.Attach(ctx)
+			ctx = newCtx
+			defer cancelFn()
+		}
+
 		defer wg.Done()
-		handler.ServeHTTP(w, req)
+		handler.ServeHTTP(w, req.WithContext(ctx))
 	})
 }
+
+type RequestTerminator interface {
```

Comment:
Can you please add comment to this interface (and Attach method) to describe what exactly is the contract of it?